pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1229,6 +1229,12 @@ func (c *Cluster) commitMaxMonIDRequireIncrementing(desiredMaxMonID int, require
 var updateDeploymentAndWait = UpdateCephDeploymentAndWait
 
 func (c *Cluster) updateMon(m *monConfig, d *apps.Deployment) error {
+
+	if c.HasMonPathChanged(m.DaemonName) {
```

Comment:
In the example output it looks like this is actually causing the failover to also be skipped. This code path is called eventually from `failoverMon()`, so we will need to know that this is failover so it shouldn't be skipped.
```
2023-10-19 11:00:31.556613 I | op-mon: fail over mon "a" to use persistent volumes
2023-10-19 11:00:31.557385 I | op-mon: mon path has changed for "a". Skip updating mon deployment in order to failover the mon
```

Reply:
fixed to logs. 

Changing from hostpath to pv:
``` 2023-10-23 05:19:55.312825 I | cephclient: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2023-10-23 05:19:55.313030 I | cephclient: generated admin config in /var/lib/rook/rook-ceph
2023-10-23 05:19:55.727259 I | op-mon: mon "b" path changed from host path to persistent volumes
2023-10-23 05:19:55.727280 I | op-mon: path has changed for mon "b". Skip updating mon deployment "rook-ceph-mon-b" in order to failover the mon
2023-10-23 05:19:55.727288 I | op-mon: waiting for mon quorum with [b c a]
2023-10-23 05:19:56.329882 I | op-mon: mons running: [b c a]
2023-10-23 05:19:56.761004 I | op-mon: Monitors in quorum: [a b c]
2023-10-23 05:19:56.771345 I | op-mon: mon "c" path changed from host path to persistent volumes
2023-10-23 05:19:56.771363 I | op-mon: path has changed for mon "c". Skip updating mon deployment "rook-ceph-mon-c" in order to failover the mon
2023-10-23 05:19:56.771370 I | op-mon: waiting for mon quorum with [b c a]
2023-10-23 05:19:56.917678 I | op-mon: mons running: [b c a]
2023-10-23 05:19:57.310862 I | op-mon: Monitors in quorum: [a b c]
2023-10-23 05:19:57.315501 I | op-mon: mon "a" path changed from host path to persistent volumes
2023-10-23 05:19:57.315520 I | op-mon: path has changed for mon "a". Skip updating mon deployment "rook-ceph-mon-a" in order to failover the mon
2023-10-23 05:19:57.315527 I | op-mon: waiting for mon quorum with [b c a]
2023-10-23 05:19:57.516714 I | op-mon: mons running: [b c a]
2023-10-23 05:19:57.979196 I | op-mon: Monitors in quorum: [a b c]
2023-10-23 05:19:57.979215 I | op-mon: mons created: 3
2023-10-23 05:19:58.388035 I | op-mon: waiting for mon quorum with [b c a]
2023-10-23 05:19:58.407239 I | op-mon: mons running: [b c a]
2023-10-23 05:19:58.829342 I | op-mon: Monitors in quorum: [a b c]
```